<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>NACHO • NACHO</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="NACHO">
<meta property="og:description" content="NACHO">
<meta name="robots" content="noindex">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">NACHO</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="In-development version">1.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/NACHO.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/NACHO-analysis.html">NACHO Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mcanouil/NACHO/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>NACHO</h1>
            <h3 class="subtitle">A NAnostring quality Control dasHbOard</h3>
                        <h4 class="author">Mickaël Canouil, Ph.D., Gerard A. Bouland and Roderick C. Slieker, Ph.D.</h4>
            
            <h4 class="date">April 27, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/mcanouil/NACHO/blob/master/vignettes/NACHO.Rmd"><code>vignettes/NACHO.Rmd</code></a></small>
      <div class="hidden name"><code>NACHO.Rmd</code></div>

    </div>

    
    
<p><img src="nacho_hex.png" width="150px" style="display: block; margin: auto;"></p>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co"># Install NACHO from CRAN:</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"NACHO"</span>)

<span class="co"># Or the the development version from GitHub:</span>
<span class="co"># install.packages("remotes")</span>
<span class="kw pkg">remotes</span><span class="kw ns">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span>(<span class="st">"mcanouil/NACHO"</span>)</pre></body></html></div>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co"># Load NACHO</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">NACHO</span>)</pre></body></html></div>
</div>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p><em>NACHO</em> (<strong>NA</strong>nostring quality <strong>C</strong>ontrol das<strong>H</strong>b<strong>O</strong>ard) is developed for NanoString nCounter data.<br>
NanoString nCounter data is a messenger-RNA/micro-RNA (mRNA/miRNA) expression assay and works with fluorescent barcodes.<br>
Each barcode is assigned a mRNA/miRNA, which can be counted after bonding with its target.<br>
As a result each count of a specific barcode represents the presence of its target mRNA/miRNA.</p>
<p><em>NACHO</em> is able to load, visualise and normalise the exported NanoString nCounter data and facilitates the user in performing a quality control.<br><em>NACHO</em> does this by visualising quality control metrics, expression of control genes, principal components and sample specific size factors in an interactive web application.</p>
<p>With the use of two functions, RCC files are summarised and visualised, namely: <code><a href="../reference/load_rcc.html">load_rcc()</a></code> and <code><a href="../reference/visualise.html">visualise()</a></code>.</p>
<ul>
<li>The <code><a href="../reference/load_rcc.html">load_rcc()</a></code> function is used to preprocess the data.</li>
<li>The <code><a href="../reference/visualise.html">visualise()</a></code> function initiates a <a href="https://shiny.rstudio.com/">Shiny-based dashboard</a> that visualises all relevant QC plots.</li>
</ul>
<p><em>NACHO</em> also includes a function <code><a href="../reference/normalise.html">normalise()</a></code>, which (re)calculates sample specific size factors and normalises the data.</p>
<ul>
<li>The <code><a href="../reference/normalise.html">normalise()</a></code> function creates a list in which your settings, the raw counts and normalised counts are stored.</li>
</ul>
<p>In addition (since v0.6.0) <em>NACHO</em> includes two (three) additional functions:</p>
<ul>
<li>The <code><a href="../reference/render.html">render()</a></code> function renders a full quality-control report (HTML) based on the results of a call to <code><a href="../reference/load_rcc.html">load_rcc()</a></code> or <code><a href="../reference/normalise.html">normalise()</a></code> (using <code><a href="https://rdrr.io/r/base/print.html">print()</a></code> in a Rmarkdown chunk).</li>
<li>The <code><a href="../reference/autoplot.html">autoplot()</a></code> function draws any quality-control metrics from <code><a href="../reference/visualise.html">visualise()</a></code> and <code><a href="../reference/render.html">render()</a></code>.</li>
</ul>
<p>For more <code><a href="../articles/NACHO.html">vignette("NACHO")</a></code> and <code><a href="../articles/NACHO-analysis.html">vignette("NACHO-analysis")</a></code>.</p>
<p>
Canouil M, Bouland GA, Bonnefond A, Froguel P, Hart L, Slieker R (2019). “NACHO: an R package for quality control of NanoString nCounter data.” <em>Bioinformatics</em>. ISSN 1367-4803, doi: <a href="https://doi.org/10.1093/bioinformatics/btz647">10.1093/bioinformatics/btz647</a>.
</p>
<pre><code>@Article{,
  title = {{NACHO}: an {R} package for quality control of {NanoString} {nCounter} data},
  author = {Mickaël Canouil and Gerard A. Bouland and Amélie Bonnefond and Philippe Froguel and Leen Hart and Roderick Slieker},
  journal = {Bioinformatics},
  address = {Oxford, England},
  year = {2019},
  month = {aug},
  issn = {1367-4803},
  doi = {10.1093/bioinformatics/btz647},
}</code></pre>
</div>
<div id="an-example" class="section level1">
<h1 class="hasAnchor">
<a href="#an-example" class="anchor"></a>An example</h1>
<p>To display the usage and utility of <em>NACHO</em>, we show three examples in which the above mentioned functions are used and the results are briefly examined.</p>
<p><em>NACHO</em> comes with presummarised data and in the first example we use this dataset to call the interactive web application using <code><a href="../reference/visualise.html">visualise()</a></code>.<br>
In the second example, we show the process of going from raw RCC files to visualisations with a dataset queried from <strong>GEO</strong> using <code>GEOquery</code>.<br>
In the third example, we use the summarised dataset from the second example to calculate the sample specific size factors using <code><a href="../reference/normalise.html">normalise()</a></code> and its added functionality to predict housekeeping genes.</p>
<p>Besides creating interactive visualisations, <em>NACHO</em> also identifies poorly performing samples which can be seen under the Outlier Table tab in the interactive web application.<br>
While calling <code><a href="../reference/normalise.html">normalise()</a></code>, the user has the possibility to remove these outliers before size factor calculation.</p>
<div id="get-nanostring-ncounter-data" class="section level2">
<h2 class="hasAnchor">
<a href="#get-nanostring-ncounter-data" class="anchor"></a>Get NanoString nCounter data</h2>
<div id="presummarised-data-from-nacho" class="section level3">
<h3 class="hasAnchor">
<a href="#presummarised-data-from-nacho" class="anchor"></a>Presummarised data from <em>NACHO</em>
</h3>
<p>This example shows how to use summarised data to call the interactive web application.<br>
The raw data used is from a study of <span class="citation">Liu et al. (<a href="#ref-liu_pam50_2016">2016</a>)</span> and was acquired from the NCBI GEO public database <span class="citation">(Barrett et al. <a href="#ref-barrett_ncbi_2013">2013</a>)</span>.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">NACHO</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">GSE74821</span>)
<span class="fu"><a href="../reference/visualise.html">visualise</a></span>(<span class="no">GSE74821</span>)</pre></body></html></div>
<p><img src="README-visualise.png" width="650px" style="display: block; margin: auto;"></p>
</div>
<div id="raw-data-from-geo" class="section level3">
<h3 class="hasAnchor">
<a href="#raw-data-from-geo" class="anchor"></a>Raw data from GEO</h3>
<p>Numerous NanoString nCounter datasets are available from GEO <span class="citation">(Barrett et al. <a href="#ref-barrett_ncbi_2013">2013</a>)</span>.<br>
In this example, we use a mRNA dataset from the study of <span class="citation">Bruce et al. (<a href="#ref-bruce_identification_2015">2015</a>)</span> with the GEO accession number: <strong>GSE70970</strong>. The data is extracted and prepared using the following code.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">GEOquery</span>)
<span class="co"># Download data</span>
<span class="no">gse</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GEOquery/man/getGEO.html">getGEO</a></span>(<span class="st">"GSE70970"</span>)
<span class="co"># Get phenotypes</span>
<span class="no">targets</span> <span class="kw">&lt;-</span> <span class="fu">pData</span>(<span class="fu">phenoData</span>(<span class="no">gse</span><span class="kw">[[</span><span class="fl">1</span>]]))
<span class="fu"><a href="https://rdrr.io/pkg/GEOquery/man/getGEOSuppFiles.html">getGEOSuppFiles</a></span>(<span class="kw">GEO</span> <span class="kw">=</span> <span class="st">"GSE70970"</span>, <span class="kw">baseDir</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>())
<span class="co"># Unzip data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/untar.html">untar</a></span>(
  <span class="kw">tarfile</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="st">"GSE70970"</span>, <span class="st">"GSE70970_RAW.tar"</span>),
  <span class="kw">exdir</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="st">"GSE70970"</span>, <span class="st">"Data"</span>)
)
<span class="co"># Add IDs</span>
<span class="no">targets</span>$<span class="no">IDFILE</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="st">"GSE70970"</span>, <span class="st">"Data"</span>))</pre></body></html></div>
<pre><code>## # A tibble: 263 x 71
##    IDFILE title geo_accession status submission_date last_update_date type 
##    &lt;chr&gt;  &lt;fct&gt; &lt;chr&gt;         &lt;fct&gt;  &lt;fct&gt;           &lt;fct&gt;            &lt;fct&gt;
##  1 GSM18… NPC-… GSM1824143    Publi… Jul 15 2015     Jul 20 2015      RNA  
##  2 GSM18… NPC-… GSM1824144    Publi… Jul 15 2015     Jul 20 2015      RNA  
##  3 GSM18… NPC-… GSM1824145    Publi… Jul 15 2015     Jul 20 2015      RNA  
##  4 GSM18… NPC-… GSM1824146    Publi… Jul 15 2015     Jul 20 2015      RNA  
##  5 GSM18… NPC-… GSM1824147    Publi… Jul 15 2015     Jul 20 2015      RNA  
##  6 GSM18… NPC-… GSM1824148    Publi… Jul 15 2015     Jul 20 2015      RNA  
##  7 GSM18… NPC-… GSM1824149    Publi… Jul 15 2015     Jul 20 2015      RNA  
##  8 GSM18… NPC-… GSM1824150    Publi… Jul 15 2015     Jul 20 2015      RNA  
##  9 GSM18… NPC-… GSM1824151    Publi… Jul 15 2015     Jul 20 2015      RNA  
## 10 GSM18… NPC-… GSM1824152    Publi… Jul 15 2015     Jul 20 2015      RNA  
## # … with 253 more rows, and 64 more variables: channel_count &lt;fct&gt;,
## #   source_name_ch1 &lt;fct&gt;, organism_ch1 &lt;fct&gt;, characteristics_ch1 &lt;fct&gt;,
## #   characteristics_ch1.1 &lt;fct&gt;, characteristics_ch1.2 &lt;fct&gt;,
## #   characteristics_ch1.3 &lt;fct&gt;, characteristics_ch1.4 &lt;fct&gt;,
## #   characteristics_ch1.5 &lt;fct&gt;, characteristics_ch1.6 &lt;fct&gt;,
## #   characteristics_ch1.7 &lt;fct&gt;, characteristics_ch1.8 &lt;fct&gt;,
## #   characteristics_ch1.9 &lt;fct&gt;, characteristics_ch1.10 &lt;fct&gt;,
## #   characteristics_ch1.11 &lt;fct&gt;, characteristics_ch1.12 &lt;fct&gt;,
## #   characteristics_ch1.13 &lt;fct&gt;, characteristics_ch1.14 &lt;fct&gt;,
## #   characteristics_ch1.15 &lt;fct&gt;, characteristics_ch1.16 &lt;fct&gt;,
## #   characteristics_ch1.17 &lt;fct&gt;, characteristics_ch1.18 &lt;fct&gt;,
## #   characteristics_ch1.19 &lt;fct&gt;, treatment_protocol_ch1 &lt;fct&gt;,
## #   growth_protocol_ch1 &lt;fct&gt;, molecule_ch1 &lt;fct&gt;, extract_protocol_ch1 &lt;fct&gt;,
## #   label_ch1 &lt;fct&gt;, label_protocol_ch1 &lt;fct&gt;, taxid_ch1 &lt;fct&gt;,
## #   hyb_protocol &lt;fct&gt;, scan_protocol &lt;fct&gt;, data_processing &lt;fct&gt;,
## #   platform_id &lt;fct&gt;, contact_name &lt;fct&gt;, contact_email &lt;fct&gt;,
## #   contact_institute &lt;fct&gt;, contact_address &lt;fct&gt;, contact_city &lt;fct&gt;,
## #   contact_state &lt;fct&gt;, `contact_zip/postal_code` &lt;fct&gt;,
## #   contact_country &lt;fct&gt;, supplementary_file &lt;fct&gt;, data_row_count &lt;fct&gt;,
## #   `age:ch1` &lt;chr&gt;, `bin.t:ch1` &lt;chr&gt;, `chemo:ch1` &lt;chr&gt;,
## #   `disease.event:ch1` &lt;chr&gt;, `disease.spec.event:ch1` &lt;chr&gt;,
## #   `disease.spec.time:ch1` &lt;chr&gt;, `disease.time:ch1` &lt;chr&gt;,
## #   `distant.event:ch1` &lt;chr&gt;, `distant.time:ch1` &lt;chr&gt;, `gender:ch1` &lt;chr&gt;,
## #   `local.event:ch1` &lt;chr&gt;, `local.regional.event:ch1` &lt;chr&gt;,
## #   `local.regional.time:ch1` &lt;chr&gt;, `local.time:ch1` &lt;chr&gt;, `n:ch1` &lt;chr&gt;,
## #   `nodal.event:ch1` &lt;chr&gt;, `nodal.time:ch1` &lt;chr&gt;,
## #   `survival.event:ch1` &lt;chr&gt;, `survival.time:ch1` &lt;chr&gt;, `t:ch1` &lt;chr&gt;</code></pre>
<p>After we extracted the dataset to the <code>/tmp/RtmpSHwMeR/GSE70970/Data</code> directory, a <code>Samplesheet.csv</code> containing a column with the exact names of the files for each sample can be written or use as is.</p>
</div>
</div>
<div id="the-load_rcc-function" class="section level2">
<h2 class="hasAnchor">
<a href="#the-load_rcc-function" class="anchor"></a>The <code>load_rcc()</code> function</h2>
<p>The first argument requires the path to the directory containing the RCC files, the second argument is the location of samplesheet followed by third argument with the column name containing the exact names of the files.<br>
The <code>housekeeping_genes</code> and <code>normalisation_method</code> arguments respectively indicate which housekeeping genes and normalisation method should be used.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">GSE70970_sum</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/load_rcc.html">load_rcc</a></span>(
  <span class="kw">data_directory</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="st">"GSE70970"</span>, <span class="st">"Data"</span>), <span class="co"># Where the data is</span>
  <span class="kw">ssheet_csv</span> <span class="kw">=</span> <span class="no">targets</span>, <span class="co"># The samplesheet</span>
  <span class="kw">id_colname</span> <span class="kw">=</span> <span class="st">"IDFILE"</span>, <span class="co"># Name of the column that contains the unique identfiers</span>
  <span class="kw">housekeeping_genes</span> <span class="kw">=</span> <span class="kw">NULL</span>, <span class="co"># Custom list of housekeeping genes</span>
  <span class="kw">housekeeping_predict</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="co"># Whether or not to predict the housekeeping genes</span>
  <span class="kw">normalisation_method</span> <span class="kw">=</span> <span class="st">"GEO"</span>, <span class="co"># Geometric mean or GLM</span>
  <span class="kw">n_comp</span> <span class="kw">=</span> <span class="fl">5</span> <span class="co"># Number indicating how many principal components should be computed. </span>
)
<span class="co">## [NACHO] Importing RCC files.</span>
<span class="co">## [NACHO] Performing QC and formatting data.</span>
<span class="co">## [NACHO] Searching for the best housekeeping genes.</span>
<span class="co">## [NACHO] Computing normalisation factors using "GEO" method for housekeeping genes prediction.</span>
<span class="co">## [NACHO] The following predicted housekeeping genes will be used for normalisation:</span>
<span class="co">##   - hsa-miR-103</span>
<span class="co">##   - hsa-let-7e</span>
<span class="co">##   - hsa-miR-1260</span>
<span class="co">##   - hsa-miR-500+hsa-miR-501-5p</span>
<span class="co">##   - hsa-miR-1274b</span>
<span class="co">## [NACHO] Computing normalisation factors using "GEO" method.</span>
<span class="co">## [NACHO] Missing values have been replaced with zeros for PCA.</span>
<span class="co">## [NACHO] Normalising data using "GEO" method with housekeeping genes.</span>
<span class="co">## [NACHO] Returning a list.</span>
<span class="co">##   $ access              : character</span>
<span class="co">##   $ housekeeping_genes  : character</span>
<span class="co">##   $ housekeeping_predict: logical</span>
<span class="co">##   $ housekeeping_norm   : logical</span>
<span class="co">##   $ normalisation_method: character</span>
<span class="co">##   $ remove_outliers     : logical</span>
<span class="co">##   $ n_comp              : numeric</span>
<span class="co">##   $ data_directory      : character</span>
<span class="co">##   $ pc_sum              : data.frame</span>
<span class="co">##   $ nacho               : data.frame</span>
<span class="co">##   $ outliers_thresholds : list</span></pre></body></html></div>
</div>
<div id="the-visualise-function" class="section level2">
<h2 class="hasAnchor">
<a href="#the-visualise-function" class="anchor"></a>The <code>visualise()</code> function</h2>
<p>When the summarisation is done, the summarised (or normalised) data can be visualised using the <code><a href="../reference/visualise.html">visualise()</a></code> function as can be seen in the following chunk of code.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="../reference/visualise.html">visualise</a></span>(<span class="no">GSE70970_sum</span>)</pre></body></html></div>
<p>The sidebar includes widgets to control quality-control thresholds. These widgets differ according to the selected tab. Each sample in the plots can be coloured based on either technical specifications which are included in the RCC files or based on specifications of your own choosing, though these specifications need to be included in the samplesheet.</p>
</div>
<div id="the-normalise-function" class="section level2">
<h2 class="hasAnchor">
<a href="#the-normalise-function" class="anchor"></a>The <code>normalise()</code> function</h2>
<p><em>NACHO</em> allows the discovery of housekeeping genes within your own dataset. <em>NACHO</em> finds the five best suitable housekeeping genes, however, it is possible that one of these five genes might not be suitable, which is why a subset of these discovered housekeeping genes might work better in some cases. For this example, we use the <strong>GSE70970</strong> dataset from the previous example. The discovered housekeeping genes are saved in the result object as <strong>predicted_housekeeping</strong>.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">GSE70970_sum</span><span class="kw">[[</span><span class="st">"housekeeping_genes"</span>]])
<span class="co">## [1] "hsa-miR-103"                "hsa-let-7e"                </span>
<span class="co">## [3] "hsa-miR-1260"               "hsa-miR-500+hsa-miR-501-5p"</span>
<span class="co">## [5] "hsa-miR-1274b"</span></pre></body></html></div>
<p>Let’s say <em>hsa-miR-103</em> and <em>hsa-let-7e</em> are not suitable, therefore, you want to exclude these genes from the normalisation process.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">my_housekeeping</span> <span class="kw">&lt;-</span> <span class="no">GSE70970_sum</span><span class="kw">[[</span><span class="st">"housekeeping_genes"</span>]][-<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">2</span>)]
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">my_housekeeping</span>)
<span class="co">## [1] "hsa-miR-1260"               "hsa-miR-500+hsa-miR-501-5p"</span>
<span class="co">## [3] "hsa-miR-1274b"</span></pre></body></html></div>
<p>The next step is the actual normalisation. The first argument requires the summary which is created with the <code><a href="../reference/load_rcc.html">load_rcc()</a></code> function. The second argument requires a vector of gene names. In this case, it is a subset of the discovered housekeeping genes we just made. With the third argument the user has the choice to remove the outliers. Lastly, the normalisation method can be choosed.<br>
Here, the user has a choice between <code>"GLM"</code> or <code>"GEO"</code>. The differences between normalisation methods are nuanced, however, a preference for either method are use case specific.<br>
In this example, <code>"GLM"</code> is used.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">GSE70970_norm</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/normalise.html">normalise</a></span>(
  <span class="kw">nacho_object</span> <span class="kw">=</span> <span class="no">GSE70970_sum</span>,
  <span class="kw">housekeeping_genes</span> <span class="kw">=</span> <span class="no">my_housekeeping</span>,
  <span class="kw">housekeeping_predict</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">housekeeping_norm</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">normalisation_method</span> <span class="kw">=</span> <span class="st">"GEO"</span>,
  <span class="kw">remove_outliers</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)
<span class="co">## [NACHO] Normalising "GSE70970_sum" with new value for parameters:</span>
<span class="co">##   - housekeeping_genes = TRUE</span>
<span class="co">##   - housekeeping_predict = TRUE</span>
<span class="co">##   - remove_outliers = TRUE</span>
<span class="co">## [NACHO] Computing normalisation factors using "GEO" method.</span>
<span class="co">## [NACHO] Missing values have been replaced with zeros for PCA.</span>
<span class="co">## [NACHO] Returning a list.</span>
<span class="co">##   $ access              : character</span>
<span class="co">##   $ housekeeping_genes  : character</span>
<span class="co">##   $ housekeeping_predict: logical</span>
<span class="co">##   $ housekeeping_norm   : logical</span>
<span class="co">##   $ normalisation_method: character</span>
<span class="co">##   $ remove_outliers     : logical</span>
<span class="co">##   $ n_comp              : numeric</span>
<span class="co">##   $ data_directory      : character</span>
<span class="co">##   $ pc_sum              : data.frame</span>
<span class="co">##   $ nacho               : data.frame</span>
<span class="co">##   $ outliers_thresholds : list</span></pre></body></html></div>
<p><code><a href="../reference/normalise.html">normalise()</a></code> returns a <code>list</code> object (same as <code><a href="../reference/load_rcc.html">load_rcc()</a></code>) with <code>raw_counts</code> and <code>normalised_counts</code> slots filled with the raw and normalised counts. Both counts are also in the <em>NACHO</em> data.frame.</p>
</div>
<div id="the-autoplot-function" class="section level2">
<h2 class="hasAnchor">
<a href="#the-autoplot-function" class="anchor"></a>The <code>autoplot()</code> function</h2>
<p>The <code><a href="../reference/autoplot.html">autoplot()</a></code> function provides an easy way to plot any quality-control from the <code><a href="../reference/visualise.html">visualise()</a></code> function.</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="../reference/autoplot.html">autoplot</a></span>(
  <span class="kw">object</span> <span class="kw">=</span> <span class="no">GSE74821</span>,
  <span class="kw">x</span> <span class="kw">=</span> <span class="st">"BD"</span>,
  <span class="kw">colour</span> <span class="kw">=</span> <span class="st">"CartridgeID"</span>,
  <span class="kw">size</span> <span class="kw">=</span> <span class="fl">0.5</span>,
  <span class="kw">show_legend</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)</pre></body></html></div>
<p>The possible metrics (<code>x</code>) are:</p>
<ul>
<li>
<code>"BD"</code> (Binding Density)</li>
<li>
<code>"FoV"</code> (Imaging)</li>
<li>
<code>"PCL"</code> (Positive Control Linearity)</li>
<li>
<code>"LoD"</code> (Limit of Detection)</li>
<li>
<code>"Positive"</code> (Positive Controls)</li>
<li>
<code>"Negative"</code> (Negative Controls)</li>
<li>
<code>"Housekeeping"</code> (Housekeeping Genes)</li>
<li>
<code>"PN"</code> (Positive Controls vs. Negative Controls)</li>
<li>
<code>"ACBD"</code> (Average Counts vs. Binding Density)</li>
<li>
<code>"ACMC"</code> (Average Counts vs. Median Counts)</li>
<li>
<code>"PCA12"</code> (Principal Component 1 vs. 2)</li>
<li>
<code>"PCAi"</code> (Principal Component scree plot)</li>
<li>
<code>"PCA"</code> (Principal Components planes)</li>
<li>
<code>"PFNF"</code> (Positive Factor vs. Negative Factor)</li>
<li>
<code>"HF"</code> (Housekeeping Factor)</li>
<li>
<code>"NORM"</code> (Normalisation Factor)</li>
</ul>
<div id="binding-density" class="section level3">
<h3 class="hasAnchor">
<a href="#binding-density" class="anchor"></a>Binding Density</h3>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="imaging" class="section level3">
<h3 class="hasAnchor">
<a href="#imaging" class="anchor"></a>Imaging</h3>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="positive-control-linearity" class="section level3">
<h3 class="hasAnchor">
<a href="#positive-control-linearity" class="anchor"></a>Positive Control Linearity</h3>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-3.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="limit-of-detection" class="section level3">
<h3 class="hasAnchor">
<a href="#limit-of-detection" class="anchor"></a>Limit of Detection</h3>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-4.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="positive-controls" class="section level3">
<h3 class="hasAnchor">
<a href="#positive-controls" class="anchor"></a>Positive Controls</h3>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-5.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="negative-controls" class="section level3">
<h3 class="hasAnchor">
<a href="#negative-controls" class="anchor"></a>Negative Controls</h3>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-6.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="housekeeping-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#housekeeping-genes" class="anchor"></a>Housekeeping Genes</h3>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-7.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="positive-controls-vs-negative-controls" class="section level3">
<h3 class="hasAnchor">
<a href="#positive-controls-vs-negative-controls" class="anchor"></a>Positive Controls vs. Negative Controls</h3>
<pre><code>## `geom_smooth()` using formula 'y ~ x'</code></pre>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-8.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="average-counts-vs-binding-density" class="section level3">
<h3 class="hasAnchor">
<a href="#average-counts-vs-binding-density" class="anchor"></a>Average Counts vs. Binding Density</h3>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-9.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="average-counts-vs-median-counts" class="section level3">
<h3 class="hasAnchor">
<a href="#average-counts-vs-median-counts" class="anchor"></a>Average Counts vs. Median Counts</h3>
</div>
<div id="principal-component-1-vs-2" class="section level3">
<h3 class="hasAnchor">
<a href="#principal-component-1-vs-2" class="anchor"></a>Principal Component 1 vs. 2</h3>
<pre><code>## Warning: `expand_scale()` is deprecated; use `expansion()` instead.

## Warning: `expand_scale()` is deprecated; use `expansion()` instead.</code></pre>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-10.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="principal-component-scree-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#principal-component-scree-plot" class="anchor"></a>Principal Component scree plot</h3>
<pre><code>## Warning: `expand_scale()` is deprecated; use `expansion()` instead.</code></pre>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-11.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="principal-components-planes" class="section level3">
<h3 class="hasAnchor">
<a href="#principal-components-planes" class="anchor"></a>Principal Components planes</h3>
<pre><code>## Warning: `expand_scale()` is deprecated; use `expansion()` instead.

## Warning: `expand_scale()` is deprecated; use `expansion()` instead.</code></pre>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-12.png" width="700" style="display: block; margin: auto;"><img src="NACHO_files/figure-html/unnamed-chunk-10-13.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="positive-factor-vs-negative-factor" class="section level3">
<h3 class="hasAnchor">
<a href="#positive-factor-vs-negative-factor" class="anchor"></a>Positive Factor vs. Negative Factor</h3>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-14.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="housekeeping-factor" class="section level3">
<h3 class="hasAnchor">
<a href="#housekeeping-factor" class="anchor"></a>Housekeeping Factor</h3>
<pre><code>## Warning: Transformation introduced infinite values in continuous x-axis
## Warning: Transformation introduced infinite values in continuous x-axis
## Warning: Transformation introduced infinite values in continuous y-axis

## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-15.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="normalisation-factor" class="section level3">
<h3 class="hasAnchor">
<a href="#normalisation-factor" class="anchor"></a>Normalisation Factor</h3>
<pre><code>## `geom_smooth()` using formula 'y ~ x'</code></pre>
<p><img src="NACHO_files/figure-html/unnamed-chunk-10-16.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="nacho-as-a-standalone-app" class="section level2">
<h2 class="hasAnchor">
<a href="#nacho-as-a-standalone-app" class="anchor"></a>NACHO as a standalone app</h2>
<p><em>NACHO</em> is also available as a standalone app to be used in a shiny server configuration. A convenience function <code><a href="../reference/deploy.html">deploy()</a></code> is available to directly copy the <em>NACHO</em> app to the default directory of a shiny server.</p>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="fu"><a href="../reference/deploy.html">deploy</a></span>(<span class="kw">directory</span> <span class="kw">=</span> <span class="st">"/srv/shiny-server"</span>, <span class="kw">app_name</span> <span class="kw">=</span> <span class="st">"NACHO"</span>)</pre></body></html></div>
<p>The app can also be run directly, without manually summarising and normalising RCC files:</p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="kw pkg">shiny</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html">runApp</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"app"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"NACHO"</span>))</pre></body></html></div>
<p><img src="README-app.png" width="650px" style="display: block; margin: auto;"></p>
</div>
<div id="the-render-function" class="section level2">
<h2 class="hasAnchor">
<a href="#the-render-function" class="anchor"></a>The <code>render()</code> function</h2>
<p>The <code><a href="../reference/render.html">render()</a></code> function renders (using <code><a href="https://rdrr.io/r/base/print.html">print(..., echo = TRUE)</a></code> a comprehensive HTML report which includes all quality-control metrics and description of those metrics.</p>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="fu"><a href="../reference/render.html">render</a></span>(
  <span class="kw">nacho_object</span> <span class="kw">=</span> <span class="no">GSE74821</span>,
  <span class="kw">colour</span> <span class="kw">=</span> <span class="st">"CartridgeID"</span>,
  <span class="kw">output_file</span> <span class="kw">=</span> <span class="st">"NACHO_QC.html"</span>,
  <span class="kw">output_dir</span> <span class="kw">=</span> <span class="st">"."</span>,
  <span class="kw">size</span> <span class="kw">=</span> <span class="fl">0.5</span>,
  <span class="kw">show_legend</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">clean</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)</pre></body></html></div>
<p>The underneath function <code><a href="https://rdrr.io/r/base/print.html">print()</a></code> can be used directly within any Rmarkdown chunk, setting the parameter <code>echo = TRUE</code>.</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(
  <span class="kw">x</span> <span class="kw">=</span> <span class="no">GSE74821</span>,
  <span class="kw">colour</span> <span class="kw">=</span> <span class="st">"CartridgeID"</span>,
  <span class="kw">size</span> <span class="kw">=</span> <span class="fl">0.5</span>,
  <span class="kw">show_legend</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">echo</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">title_level</span> <span class="kw">=</span> <span class="fl">3</span>
)</pre></body></html></div>
<div id="rcc-summary" class="section level3">
<h3 class="hasAnchor">
<a href="#rcc-summary" class="anchor"></a>RCC Summary</h3>
<ul>
<li>Samples: 48</li>
<li>Endogenous: 50</li>
<li>Housekeeping: 8</li>
<li>Negative: 8</li>
<li>Positive: 6</li>
</ul>
</div>
<div id="settings" class="section level3">
<h3 class="hasAnchor">
<a href="#settings" class="anchor"></a>Settings</h3>
<ul>
<li>Predict housekeeping genes: FALSE</li>
<li>Normalise using housekeeping genes: TRUE</li>
<li>Housekeeping genes available: MRPL19, PSMC4, SF3A1, RPLP0, PUM1, ACTB, TFRC and GUSB</li>
<li>Normalise using: GLM</li>
<li>Principal components to compute: 10</li>
<li>
<p>Remove outliers: FALSE</p>
<ul>
<li>Binding Density (BD) &lt; 0.1</li>
<li>Binding Density (BD) &gt; 2.25</li>
<li>Field of View (FoV) &lt; 75</li>
<li>Positive Control Linearity (PCL) &lt; 0.95</li>
<li>Limit of Detection (LoD) &lt; 2</li>
<li>Positive normalisation factor (Positive_factor) &lt; 0.25</li>
<li>Positive normalisation factor (Positive_factor) &gt; 4</li>
<li>Housekeeping normalisation factor (house_factor) &lt; 0.091</li>
<li>Housekeeping normalisation factor (house_factor) &gt; 11</li>
</ul>
</li>
</ul>
</div>
<div id="qc-metrics" class="section level3">
<h3 class="hasAnchor">
<a href="#qc-metrics" class="anchor"></a>QC Metrics</h3>
<div id="binding-density-1" class="section level4">
<h4 class="hasAnchor">
<a href="#binding-density-1" class="anchor"></a>Binding Density</h4>
<p>The imaging unit only counts the codes that are unambiguously distinguishable.<br>
It simply will not count codes that overlap within an image.<br>
This provides increased confidence that the molecular counts you receive are from truly recognisable codes.<br>
Under most conditions, forgoing the few barcodes that do overlap will not impact your data.<br>
Too many overlapping codes in the image, however, will create a condition called image saturation in which significant data loss could occur (critical data loss from saturation is uncommon).</p>
<p>To determine the level of image saturation, the nCounter instrument calculates the number of optical features per square micron for each lane as it processes the images.<br>
This is called the <strong>Binding Density</strong> (<strong>BD</strong>).<br>
The <strong>Binding Density</strong> is useful for determining whether data collection has been compromised due to image saturation. The acceptable range for <strong>Binding Density</strong> is:</p>
<ul>
<li>
<code>0.1 - 2.25</code> for <strong>MAX</strong>/<strong>FLEX</strong> instruments</li>
<li>
<code>0.1 - 1.8</code> for <strong>SPRINT</strong> instruments</li>
</ul>
<p>Within these ranges, relatively few reporters on the slide surface will overlap, enabling the instrument to accurately tabulate counts for each reporter species.<br>
A <strong>Binding Density</strong> significantly greater than the upper limit in either range is indicative of overlapping reporters on the slide surface.<br>
The counts observed in lanes with a <strong>Binding Density</strong> at this level may have had significant numbers of codes ignored, which could potentially affect quantification and linearity of the assay.</p>
<p><img src="NACHO_files/figure-html/print-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="field-of-view-imaging" class="section level4">
<h4 class="hasAnchor">
<a href="#field-of-view-imaging" class="anchor"></a>Field of View (Imaging)</h4>
<p>Each individual lane scanned on an nCounter system is divided into a few hundred imaging sections, called Fields of View (<strong>FOV</strong>), the exact number of which will depend on the system being used (<em>i.e.</em>, <strong>MAX/FLEX</strong> or <strong>SPRINT</strong>), and the scanner settings selected by the user.<br>
The system images these <strong>FOV</strong>s separately, and sums the barcode counts of all <strong>FOV</strong>s from a single lane to form the final raw data count for each unique barcode target.<br>
Finally, the system reports the number of <strong>FOV</strong>s successfully imaged as <strong>FOV</strong> Counted.</p>
<p>Significant discrepancy between the number of <strong>FOV</strong> for which imaging was attempted (<strong>FOV Count</strong>) and for which imaging was successful (<strong>FOV Counted</strong>) may indicate an issue with imaging performance.<br>
Recommended percentage of registered FOVs (<em>i.e.</em>, <strong>FOV Counted</strong> over <strong>FOV Count</strong>) is <code>75 %</code>.</p>
<p><img src="NACHO_files/figure-html/print-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="positive-control-linearity-1" class="section level4">
<h4 class="hasAnchor">
<a href="#positive-control-linearity-1" class="anchor"></a>Positive Control Linearity</h4>
<p>Six synthetic DNA control targets are included with every nCounter Gene Expression assay.<br>
Their concentrations range linearly (in <em>codeset</em>) from <code>128 fM</code> to <code>0.125 fM</code>, and they are referred to as <strong>POS_A</strong> to <strong>POS_F</strong>, respectively.<br>
These <strong>Positive Controls</strong> are typically used to measure the efficiency of the hybridization reaction, and their step-wise concentrations also make them useful in checking the linearity performance of the assay.</p>
<p>Since the known concentrations of the <strong>Positive Controls</strong> increase in a linear fashion, the resulting counts should, as well.</p>
<!--
Note that because **POS_F** has a known concentration of `0.125 fM`, which is considered below the limit of detection of the system, it should be excluded from this calculation (although you will see that **POS_F** counts are significantly higher than the negative control counts in most cases).
-->
<p><img src="NACHO_files/figure-html/print-3.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="limit-of-detection-1" class="section level4">
<h4 class="hasAnchor">
<a href="#limit-of-detection-1" class="anchor"></a>Limit of Detection</h4>
<p>The limit of detection (<strong>LoD</strong>) is determined by measuring the ability to detect <strong>POS_E</strong>, the <code>0.5 fM</code> positive control probe, which corresponds to about 10,000 copies of this target within each sample tube.<br>
On a <strong>FLEX</strong>/<strong>MAX</strong> system, the standard input of <code>100 ng</code> of total RNA will roughly correspond to about 10,000 cell equivalents (assuming one cell contains <code>10 pg</code> total RNA on average).<br>
An nCounter assay run on the <strong>FLEX</strong>/<strong>MAX</strong> system should thus conservatively be able to detect roughly one transcript copy per cell for each target (or 10,000 total transcript copies).<br>
In most (codeset) assays, you will observe that even the <strong>POS_F</strong> probe (equivalent to 0.25 copies per cell) is detectable above background.</p>
<p><img src="NACHO_files/figure-html/print-4.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="control-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#control-genes" class="anchor"></a>Control Genes</h3>
<div id="positive" class="section level4">
<h4 class="hasAnchor">
<a href="#positive" class="anchor"></a>Positive</h4>
<p><img src="NACHO_files/figure-html/print-5.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="negative" class="section level4">
<h4 class="hasAnchor">
<a href="#negative" class="anchor"></a>Negative</h4>
<p><img src="NACHO_files/figure-html/print-6.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="housekeeping" class="section level4">
<h4 class="hasAnchor">
<a href="#housekeeping" class="anchor"></a>Housekeeping</h4>
<p><img src="NACHO_files/figure-html/print-7.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="control-probe-expression" class="section level4">
<h4 class="hasAnchor">
<a href="#control-probe-expression" class="anchor"></a>Control Probe Expression</h4>
<pre><code>## `geom_smooth()` using formula 'y ~ x'</code></pre>
<p><img src="NACHO_files/figure-html/print-8.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="qc-visuals" class="section level3">
<h3 class="hasAnchor">
<a href="#qc-visuals" class="anchor"></a>QC Visuals</h3>
<div id="average-count-vs-binding-density" class="section level4">
<h4 class="hasAnchor">
<a href="#average-count-vs-binding-density" class="anchor"></a>Average Count vs. Binding Density</h4>
<p><img src="NACHO_files/figure-html/print-9.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="average-count-vs-median-count" class="section level4">
<h4 class="hasAnchor">
<a href="#average-count-vs-median-count" class="anchor"></a>Average Count vs. Median Count</h4>
<p><img src="NACHO_files/figure-html/print-10.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="principal-component" class="section level4">
<h4 class="hasAnchor">
<a href="#principal-component" class="anchor"></a>Principal Component</h4>
<div id="pc1-vs-pc2" class="section level5">
<h5 class="hasAnchor">
<a href="#pc1-vs-pc2" class="anchor"></a>PC1 vs. PC2</h5>
<p><img src="NACHO_files/figure-html/print-11.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="factorial-planes" class="section level5">
<h5 class="hasAnchor">
<a href="#factorial-planes" class="anchor"></a>Factorial planes</h5>
<p><img src="NACHO_files/figure-html/print-12.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="inertia" class="section level5">
<h5 class="hasAnchor">
<a href="#inertia" class="anchor"></a>Inertia</h5>
<p><img src="NACHO_files/figure-html/print-13.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
</div>
<div id="normalisation-factors" class="section level3">
<h3 class="hasAnchor">
<a href="#normalisation-factors" class="anchor"></a>Normalisation Factors</h3>
<div id="positive-factor-vs-background-threshold" class="section level4">
<h4 class="hasAnchor">
<a href="#positive-factor-vs-background-threshold" class="anchor"></a>Positive Factor vs. Background Threshold</h4>
<p><img src="NACHO_files/figure-html/print-14.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="housekeeping-factor-1" class="section level4">
<h4 class="hasAnchor">
<a href="#housekeeping-factor-1" class="anchor"></a>Housekeeping Factor</h4>
<p><img src="NACHO_files/figure-html/print-15.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="normalisation-result" class="section level4">
<h4 class="hasAnchor">
<a href="#normalisation-result" class="anchor"></a>Normalisation Result</h4>
<pre><code>## `geom_smooth()` using formula 'y ~ x'</code></pre>
<p><img src="NACHO_files/figure-html/print-16.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-barrett_ncbi_2013">
<p>Barrett, Tanya, Stephen E. Wilhite, Pierre Ledoux, Carlos Evangelista, Irene F. Kim, Maxim Tomashevsky, Kimberly A. Marshall, et al. 2013. “NCBI GEO: Archive for Functional Genomics Data Sets—Update.” <em>Nucleic Acids Research</em> 41 (D1): D991–D995. <a href="https://doi.org/10.1093/nar/gks1193" class="uri">https://doi.org/10.1093/nar/gks1193</a>.</p>
</div>
<div id="ref-bruce_identification_2015">
<p>Bruce, Jeff P., Angela B. Y. Hui, Wei Shi, Bayardo Perez-Ordonez, Ilan Weinreb, Wei Xu, Benjamin Haibe-Kains, et al. 2015. “Identification of a microRNA Signature Associated with Risk of Distant Metastasis in Nasopharyngeal Carcinoma.” <em>Oncotarget</em> 6 (6): 4537–50. <a href="https://doi.org/10.18632/oncotarget.3005" class="uri">https://doi.org/10.18632/oncotarget.3005</a>.</p>
</div>
<div id="ref-liu_pam50_2016">
<p>Liu, Minetta C., Brandelyn N. Pitcher, Elaine R. Mardis, Sherri R. Davies, Paula N. Friedman, Jacqueline E. Snider, Tammi L. Vickery, et al. 2016. “PAM50 Gene Signatures and Breast Cancer Prognosis with Adjuvant Anthracycline- and Taxane-Based Chemotherapy: Correlative Analysis of C9741 (Alliance).” <em>Npj Breast Cancer</em> 2 (January): 15023. <a href="https://doi.org/10.1038/npjbcancer.2015.23" class="uri">https://doi.org/10.1038/npjbcancer.2015.23</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="http://mickael.canouil.fr">Mickaël Canouil</a>, Roderick Slieker, Gerard Bouland.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
