<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Render a HTML report from <a href='summarise.html'>summarise</a> or <a href='normalise.html'>normalise</a> — render • NACHO</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Render a HTML report from <a href='summarise.html'>summarise</a> or <a href='normalise.html'>normalise</a> — render" />

<meta property="og:description" content="This function create a Rmarkdown script and render it as a HTML document.
The HTML document is a quality-control report using all the metrics from visualise
based on recommendations from NanoString.
This function allows to print text and figures from the results of a call to summarise()
or normalise().
It is intended to be used in a Rmarkdown chunk." />
<meta name="twitter:card" content="summary" />

<meta name="robots" content="noindex">

<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">NACHO</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="In-development version">0.6.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/NACHO.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mcanouil/NACHO">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Render a HTML report from <a href='summarise.html'>summarise</a> or <a href='normalise.html'>normalise</a></h1>
    <small class="dont-index">Source: <a href='https://github.com/mcanouil/NACHO/blob/master/R/render.R'><code>R/render.R</code></a></small>
    <div class="hidden name"><code>render.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function create a <code>Rmarkdown</code> script and render it as a HTML document.
The HTML document is a quality-control report using all the metrics from <a href='visualise.html'>visualise</a>
based on recommendations from NanoString.</p>
<p>This function allows to print text and figures from the results of a call to <code><a href='summarise.html'>summarise()</a></code>
or <code><a href='normalise.html'>normalise()</a></code>.
It is intended to be used in a <code>Rmarkdown</code> chunk.</p>
    
    </div>

    <pre class="usage"><span class='fu'>render</span>(<span class='no'>nacho_object</span>, <span class='kw'>colour</span> <span class='kw'>=</span> <span class='st'>"CartridgeID"</span>,
  <span class='kw'>output_file</span> <span class='kw'>=</span> <span class='st'>"NACHO_QC.html"</span>, <span class='kw'>output_dir</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/getwd'>getwd</a></span>(), <span class='kw'>size</span> <span class='kw'>=</span> <span class='fl'>0.5</span>,
  <span class='kw'>show_legend</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>keep_rmd</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)

<span class='fu'>print_nacho</span>(<span class='no'>nacho_object</span>, <span class='kw'>colour</span> <span class='kw'>=</span> <span class='st'>"CartridgeID"</span>, <span class='kw'>size</span> <span class='kw'>=</span> <span class='fl'>0.5</span>,
  <span class='kw'>show_legend</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>nacho_object</th>
      <td><p>[<a href='https://www.rdocumentation.org/packages/base/topics/list'>list</a>] List obtained from <a href='summarise.html'>summarise</a> or <a href='normalise.html'>normalise</a>.</p></td>
    </tr>
    <tr>
      <th>colour</th>
      <td><p>[<a href='https://www.rdocumentation.org/packages/base/topics/character'>character</a>] Character string of the column in <code>ssheet_csv</code>
or more generally in <code>nacho_object$nacho</code> to be used as grouping colour.</p></td>
    </tr>
    <tr>
      <th>output_file</th>
      <td><p>[<a href='https://www.rdocumentation.org/packages/base/topics/character'>character</a>] The name of the output file.
If using <code>NULL</code> then the output filename will be based on filename for the input file.
If a filename is provided, a path to the output file can also be provided.
Note that the <code>output_dir</code> option allows for specifying the output file path as well,
however, if also specifying the path, the directory must exist.
If <code>output_file</code> is specified but does not have a file extension,
an extension will be automatically added according to the output format.
To avoid the automatic file extension, put the <code>output_file</code> value in <code><a href='https://www.rdocumentation.org/packages/base/topics/AsIs'>I()</a></code>, <em>e.g.</em>, <code><a href='https://www.rdocumentation.org/packages/base/topics/AsIs'>I('my-output')</a></code>.</p></td>
    </tr>
    <tr>
      <th>output_dir</th>
      <td><p>[<a href='https://www.rdocumentation.org/packages/base/topics/character'>character</a>] The output directory for the rendered output_file.
This allows for a choice of an alternate directory to which the output file should be written
(the default output directory is the working directory, <em>i.e.</em>, <code><a href='https://www.rdocumentation.org/packages/base/topics/getwd'>getwd()</a></code>).
If a path is provided with a filename in <code>output_file</code> the directory specified here will take precedence.
Please note that any directory path provided will create any necessary directories if they do not exist.</p></td>
    </tr>
    <tr>
      <th>size</th>
      <td><p>[<a href='https://www.rdocumentation.org/packages/base/topics/numeric'>numeric</a>] A <code>numeric</code> controlling point size (geom_point or geom_beeswarm)
or line size (geom_line).</p></td>
    </tr>
    <tr>
      <th>show_legend</th>
      <td><p>[<a href='https://www.rdocumentation.org/packages/base/topics/logical'>logical</a>] Boolean to indicate whether the plot legends should
be plotted (<code>TRUE</code>) or not (<code>FALSE</code>). Default is <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>keep_rmd</th>
      <td><p>[<a href='https://www.rdocumentation.org/packages/base/topics/logical'>logical</a>] Boolean to indicate whether the Rmd file used to produce the HTML report
is copied to the directory provided in <code>output_dir</code>. Default is <code>FALSE</code>.</p></td>
    </tr>
    </table>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>GSE74821</span>)
<span class='fu'>render</span>(<span class='kw'>nacho_object</span> <span class='kw'>=</span> <span class='no'>GSE74821</span>, <span class='kw'>output_dir</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</div><div class='output co'>#&gt; <span class='warning'>Warning: Transformation introduced infinite values in continuous y-axis</span></div><div class='output co'>#&gt; <span class='error'>Error: pandoc document conversion failed with error 1</span></div><div class='input'>

<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>GSE74821</span>)
<span class='fu'>print_nacho</span>(<span class='no'>GSE74821</span>, <span class='kw'>colour</span> <span class='kw'>=</span> <span class='st'>"CartridgeID"</span>, <span class='kw'>size</span> <span class='kw'>=</span> <span class='fl'>0.5</span>, <span class='kw'>show_legend</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; 
#&gt; 
#&gt; # RCC Summary
#&gt; 
#&gt;   - Samples: 20 
#&gt;   - Endogenous: 50
#&gt;    - Housekeeping: 8
#&gt;    - Negative: 8
#&gt;    - Positive: 6
#&gt; 
#&gt; 
#&gt; # Settings
#&gt; 
#&gt;   - Predict housekeeping genes: FALSE 
#&gt;   - Normalise using housekeeping genes: TRUE 
#&gt;   - Housekeeping genes available: MRPL19, PSMC4, SF3A1, RPLP0, PUM1, ACTB, TFRC and GUSB 
#&gt;   - Normalise using: GLM 
#&gt;   - Principal components to compute: 10 
#&gt;   - Remove outliers: FALSE 
#&gt; 
#&gt;      +  Binding Density (BD) &lt; 0.1 
#&gt;      +  Binding Density (BD) &gt; 2.25 
#&gt;      +  Imaging (FoV) &lt; 75 
#&gt;      +  Positive Control Linearity (PC) &lt; 0.95 
#&gt;      +  Limit of Detection (LoD) &lt; 2 
#&gt;      +  Positive normalisation factor (Positive_factor) &lt; 0.25 
#&gt;      +  Positive normalisation factor (Positive_factor) &gt; 4 
#&gt;      +  Housekeeping normalisation factor (house_factor) &lt; 0.091 
#&gt;      +  Housekeeping normalisation factor (house_factor) &gt; 11 
#&gt; 
#&gt; 
#&gt; # QC Metrics
#&gt; 
#&gt; 
#&gt; 
#&gt; ## Binding Density 
#&gt; 
#&gt; The imaging unit only counts the codes that are unambiguously distinguishable.
#&gt; It simply will not count codes that overlap within an image.
#&gt; This provides increased confidence that the molecular counts you receive are from truly recognisable codes.
#&gt; Under most conditions, forgoing the few barcodes that do overlap will not impact your data.
#&gt; Too many overlapping codes in the image, however, will create a condition called image saturation in which significant data loss could occur (critical data loss from saturation is uncommon).
#&gt; 
#&gt; 
#&gt; To determine the level of image saturation, the nCounter instrument calculates the number of optical features per square micron for each lane as it processes the images.
#&gt; This is called the **Binding Density**.
#&gt; The **Binding Density** is useful for determining whether data collection has been compromised due to image saturation.
#&gt; The acceptable range for **Binding Density** is:
#&gt; 
#&gt; 
#&gt; * `0.1 - 2.25` for **MAX**/**FLEX** instruments
#&gt; * `0.1 - 1.8` for **SPRINT** instruments
#&gt; 
#&gt; 
#&gt; Within these ranges, relatively few reporters on the slide surface will overlap, enabling the instrument to accurately tabulate counts for each reporter species.
#&gt; A **Binding Density** significantly greater than the upper limit in either range is indicative of overlapping reporters on the slide surface.
#&gt; The counts observed in lanes with a **Binding Density** at this level may have had significant numbers of codes ignored, which could potentially affect quantification and linearity of the assay.
#&gt; Some of the factors that may contribute to increased **Binding Density** are listed in the Factors affecting **Binding Density** box. </div><div class='img'><img src='render-1.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ## Field of View (Imaging) 
#&gt; 
#&gt; Each individual lane scanned on an nCounter system is divided into a few hundred imaging sections, called Fields of View (**FOV**), the exact number of which will depend on the system being used (*i.e.*, **MAX/FLEX** or **SPRINT**), and the scanner settings selected by the user.
#&gt; The system images these FOVs separately, and sums the barcode counts of all **FOV**s from a single lane to form the final raw data count for each unique barcode target.
#&gt; Finally, the system reports the number of **FOV**s successfully imaged as FOV Counted.
#&gt; 
#&gt; 
#&gt; Significant discrepancy between the number of **FOV** for which imaging was attempted (**FOV Count**) and for which imaging was successful (**FOV Counted**) may indicate an issue with imaging performance.
#&gt; Recommended percentage of registered FOVs (*i.e.*, **FOV Counted** over **FOV Count**) is `75 %`.
#&gt; Lanes will be flagged if this percentage is lower. </div><div class='img'><img src='render-2.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ## Positive Control linearity 
#&gt; 
#&gt; Six synthetic DNA control targets are included with every nCounter Gene Expression assay.
#&gt; Their concentrations range linearly from `128 fM` to `0.125 fM`, and they are referred to as **POS_A** to **POS_F**, respectively.
#&gt; These **Positive Controls** are typically used to measure the efficiency of the hybridization reaction, and their step-wise concentrations also make them useful in checking the linearity performance of the assay.
#&gt; An R2 value is calculated from the regression between the known concentration of each of the **Positive Controls** and the resulting counts from them (this calculation is performed using log2-transformed values).
#&gt; 
#&gt; 
#&gt; Since the known concentrations of the **Positive Controls** increase in a linear fashion, the resulting counts should, as well.
#&gt; Therefore, R2 values should be higher than `0.95`.
#&gt; 
#&gt; 
#&gt; Note that because POS_F has a known concentration of `0.125 fM`, which is considered below the limit of detection of the system, it should be excluded from this calculation (although you will see that **POS_F** counts are significantly higher than the negative control counts in most cases). </div><div class='img'><img src='render-3.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ## Limit of Detection 
#&gt; 
#&gt; The limit of detection is determined by measuring the ability to detect **POS_E**, the `0.5 fM` positive control probe, which corresponds to about 10,000 copies of this target within each sample tube.
#&gt; On a **FLEX**/**MAX** system, the standard input of `100 ng` of total RNA will roughly correspond to about 10,000 cell equivalents (assuming one cell contains `10 pg` total RNA on average).
#&gt; An nCounter assay run on the **FLEX**/**MAX** system should thus conservatively be able to detect roughly one transcript copy per cell for each target (or 10,000 total transcript copies).
#&gt; In most assays, you will observe that even the **POS_F** probe (equivalent to 0.25 copies per cell) is detectable above background.
#&gt; 
#&gt; 
#&gt; To determine whether **POS_E** is detectable, it can be compared to the counts for the negative control probes.
#&gt; Every nCounter Gene Expression assay is manufactured with eight negative control probes that should not hybridize to any targets within the sample.
#&gt; Counts from these will approximate general non-specific binding of probes within the samples being run.
#&gt; The counts of **POS_E** should be higher than two times the standard deviation above the mean of the negative control. </div><div class='img'><img src='render-4.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; # Control Genes
#&gt; 
#&gt; 
#&gt; 
#&gt; ## Positive 
#&gt; </div><div class='img'><img src='render-5.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ## Negative 
#&gt; </div><div class='img'><img src='render-6.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ## Housekeeping 
#&gt; </div><div class='img'><img src='render-7.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ## Control Probe Expression
#&gt; </div><div class='output co'>#&gt; <span class='warning'>Warning: Transformation introduced infinite values in continuous y-axis</span></div><div class='img'><img src='render-8.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; # QC Visuals
#&gt; 
#&gt; 
#&gt; 
#&gt; ## Average Count vs. Binding Density
#&gt; </div><div class='img'><img src='render-9.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ## Average Count vs. Median Count
#&gt; </div><div class='img'><img src='render-10.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ## Principal Component
#&gt; 
#&gt; 
#&gt; 
#&gt; ### PC1 vs. PC2
#&gt; </div><div class='img'><img src='render-11.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ### Factorial planes
#&gt; </div><div class='img'><img src='render-12.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ### Inertia
#&gt; </div><div class='img'><img src='render-13.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; # Normalisation Factors
#&gt; 
#&gt; 
#&gt; 
#&gt; ## Positive Factor vs. Background Threshold
#&gt; </div><div class='img'><img src='render-14.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ## Housekeeping Factor
#&gt; </div><div class='img'><img src='render-15.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; 
#&gt; 
#&gt; 
#&gt; ## Normalisation Result
#&gt; </div><div class='img'><img src='render-16.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; </div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
            
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Mickaël Canouil, Roderick Slieker, Gerard Bouland.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

